services/src/main/java/org/keycloak/services/resources/admin/UsersResource.java

```diff
@@ -169,10 +169,10 @@ public Response createUser(final UserRepresentation rep) {
             RepresentationToModel.createGroups(rep, realm, user);
 
             RepresentationToModel.createCredentials(rep, session, realm, user, true);
-            adminEvent.operation(OperationType.CREATE).resourcePath(session.getContext().getUri(), user.getId()).representation(rep).success();
 
             if (session.getTransactionManager().isActive()) {
                 session.getTransactionManager().commit();
+                adminEvent.operation(OperationType.CREATE).resourcePath(session.getContext().getUri(), user.getId()).representation(rep).success();
```

Comment:
Need to move this to below the if statement, as it could be the tx is already committed, which would mean the event wouldn't be sent.